### ISSUE TITLE
IPictureService.GetPictureUrl method should return relative or absolute paths (based on a parameter)

### ISSUE THREAD

**Issue by mariannk** [nopSolutions/nopCommerce#908]
IPictureService.GetPictureUrl method should return relative or absolute paths (based on a parameter). 

BTW, In this case we could simplify caching of picture models in CatalogController. 

Note that last years it's better practice to have absolute URLs 
Open PictureService.cs and add 
protected virtual string GetImageUrlPrefix(string storeLocation) 
{ 
var imageUrlPrefix = ""; 
if (!String.IsNullOrEmpty(storeLocation)) 
{ 
imageUrlPrefix = storeLocation; 
} 
else 
{ 
//imageUrlPrefix = VirtualPathUtility.ToAppRelative("~/"); 
imageUrlPrefix = _httpContext.Request.ApplicationPath; 
} 
return imageUrlPrefix; 
} 
P.S. In virtual category add / to the after imageUrlPrefix = _httpContext.Request.ApplicationPath; 
then replace 

storeLocation = !String.IsNullOrEmpty(storeLocation)

```
? storeLocation
                            : _webHelper.GetStoreLocation();
```

with 

GetImageUrlPrefix(storeLocation) 

then remove pciture caching by "is connection SSL secured?" and "current store ID" 

do not forget about images for Open Graph, Twitter meta tags

**Comment by AndreiMaz** [comment]
After some discussion decided not to implement it out of the box. We leave it for customization or plugin development

**Comment by RomanovM** [comment]
It may help with solving the problem described here https://www.nopcommerce.com/boards/t/52431/images-problem-40-version.aspx

**Comment by skoshelev** [comment]
Closed #908
